cpp/src/io/json/json_column.cu

```diff
@@ -540,6 +577,12 @@ void make_device_json_column(device_span<SymbolT const> input,
     col.column_order.clear();
   };
 
+  path_from_tree tree_path{column_categories,
```

Comment:
The object is light weight. It holds span and couple of primitives. So, it may not matter much if the suggestion is due to reduce runtime or memory.
I added the struct because in future it can have a memoizer.